streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -77,6 +77,166 @@ protected Topology(final InternalTopologyBuilder internalTopologyBuilder) {
     public enum AutoOffsetReset {
         EARLIEST, LATEST
     }
+    /**
+     * Adds a new source that consumes the specified topics and forwards the records to child processor and/or sink nodes.
+     * The source will use the specified {@link org.apache.kafka.streams.AutoOffsetReset offset reset policy} if no committed offsets are found.
+     *
+     * @param offsetReset the auto offset reset policy to use for this source if no committed offsets are found; acceptable values: earliest or latest
+     * @param name the unique name of the source used to reference this node when {@link #addProcessor(String, ProcessorSupplier, String...) adding processor children}
+     * @param topics the name of one or more Kafka topics that this source is to consume
+     * @return itself
+     * @throws TopologyException if a processor is already added or if topics have already been registered by another source
+     */
+    public synchronized Topology addSource(final org.apache.kafka.streams.AutoOffsetReset offsetReset,
+                                           final String name,
+                                           final String... topics) {
+        internalTopologyBuilder.addSource(null, name, null, null, null, topics);
```

Comment:
We need to pass `offsetRest` to `addSource(...)` -- similar below, and thus also update `InternalTopologyBuilder` itself.

Reply:
Thank you, updated this

Reply:
I don't see an update yet